jkube-kit/config/service/src/main/java/org/eclipse/jkube/kit/config/service/JKubeServiceHub.java

```diff
@@ -181,4 +185,12 @@ public KubernetesClient getClient() {
     public ClusterAccess getClusterAccess() {
         return clusterAccessLazyBuilder.get();
     }
+
+    private IngressControllerDetectorService createIngressControllerDetectorService() {
+        final PluginServiceFactory<KubernetesClient> pluginServiceFactory = new PluginServiceFactory<>(getClient());
+        List<IngressControllerDetector> ingressControllerDetectors = pluginServiceFactory.createServiceObjects("META-INF/jkube/ingress-detectors");
```

Comment:
Naive question: What is the value of retrieving the list of detectors in a file, versus providing the list directly in a static field ?

Reply:
One benefit I can think of using PluginServiceFactory is that we can specify priorities of different Service implementations (in case we want this in future)

Reply:
If it detects one of the list, it won't display the message ... the priority is not important here in IMHO

Reply:
I can revert it back to what it was before, but need to check with @manusa first since he suggested using PluginServiceFactory.

Reply:
>  What is the value of retrieving the list of detectors in a file

- Consistency with the rest of components (Generators, Enrichers, and so on)
- Configuration is easier to override than code
- Scalability
- Users able to provide their own implementations
- inversion of control in general
- ...

Since we have this feature, it wouldn't be smart not taking advantage of it